dockers/Screenshotter/screenshotter.js

```diff
@@ -273,6 +276,64 @@ function imageDimensions(img) {
 }
 
 //////////////////////////////////////////////////////////////////////
+// Work out how to connect to host KaTeX server
+
+function findHostIP() {
+    if (!katexIP) {
+        katexIP = "localhost";
+    }
+    if (katexIP !== "*any*" || katexURL) {
+        if (!katexURL) {
+            katexURL = "http://" + katexIP + ":" + katexPort + "/";
+            console.log("KaTeX URL is " + katexURL);
+        }
+        process.nextTick(takeScreenshots);
+        return;
+    }
+
+    // Now we need to find an IP the container can connect to.
+    // First, install a server component to get notified of successful connects
+    app.get("/ss-connect.js", function(req, res, next) {
+        if (!katexURL) {
+            katexIP = req.query.ip;
+            katexURL = "http://" + katexIP + ":" + katexPort + "/";
+            console.log("KaTeX URL is " + katexURL);
+            process.nextTick(takeScreenshots);
+        }
+        res.setHeader("Content-Type", "text/javascript");
+        res.send("//OK");
+    });
+
+    // Next, enumerate all network addresses
+    var ips = [];
+    var devs = os.networkInterfaces();
+    for (var dev in devs) {
+        if (devs.hasOwnProperty(dev)) {
+            var addrs = devs[dev];
+            for (var i = 0; i < addrs.length; ++i) {
+                var addr = addrs[i].address;
+                if (/:/.test(addr)) {
+                    addr = "[" + addr + "]";
+                }
+                ips.push(addr);
+            }
+        }
+    }
+    console.log("Looking for host IP among " + ips.join(", "));
+
+    // Load a data: URI document which attempts to contact each of these IPs
+    var html = "<!doctype html>\n<html><body>\n";
+    html += ips.map(function(ip) {
+        return '<script src="http://' + ip + ':' + katexPort +
+            '/ss-connect.js?ip=' + encodeURIComponent(ip) +
+            '" defer></script>';
+    }).join("\n");
```

Comment:
wow... I'm surprised there isn't an easier way/some library that already does this.